### PR TITLE
Draft new Certs PR

### PR DESCRIPTION
**Pull request by DanielBelmes**
Added support for mkcert script and ensured that all our servers pick up scripts through .env.local

closes https://github.com/EtherealEngine/etherealengine/issues/8958
https://github.com/EtherealEngine/etherealengine/issues/9009

### CODE REVIEW

**Review by HexaField — COMMENTED** [review]
local.storage.ts is where server-local-files is called from. we should remove the npm script and create the https server in the storage provider constructor directly, allowing us to use the correct cert destination

File: .env.local.default [hunk 1]
```diff
@@ -15,7 +15,7 @@ KEY=certs/key.pem
 
 
 # Client variables ---------------
-APP_TITLE=Ethereal Engine
+APP_TITLE="Ethereal Engine"
```
Comment (HexaField) [inline .env.local.default:18]: please revert these .env.local.default values

**Review by HexaField — CHANGES_REQUESTED** [review]

**Comment by hanzlamateen** [comment]
Is there a readme on how to use this?

File: .env.local.default (hunk 1 above)
Comment (DanielBelmes) [reply .env.local.default:18]: I can't do so. It's not a valid bash env variable. I can also escape the space as an alternative

**Comment by DanielBelmes** [comment]
@hanzlamateen ill make a quick one today.

**Comment by DanielBelmes** [comment]
@hanzlamateen 

1. Install mkcert https://github.com/FiloSottile/mkcert
2. Change .env.local CERT and KEY to `/certs/mkcert-cert.pem /certs/mkcert-key.pem`
3. npm run dev-gencerts(Will use VITE_APP_HOST as the domain for the cert)
4. Run npm run dev and enjoy a secure context

**Comment by hanzlamateen** [comment]
> @hanzlamateen
> 
> 1. Install mkcert https://github.com/FiloSottile/mkcert
> 2. Change .env.local CERT and KEY to `/certs/mkcert-cert.pem /certs/mkcert-key.pem`
> 3. npm run dev-gencerts(Will use VITE_APP_HOST as the domain for the cert)
> 4. Run npm run dev and enjoy a secure context

I tried following above steps and then tried opening engine in a new Edge browser window. It give me certificate error. Am I missing something?

image
image

File: .env.local.default (hunk 1 above)
Comment (DanielBelmes) [reply .env.local.default:18]: Alright I have confirmed it does not change any exiting behavior

**Comment by DanielBelmes** [comment]
> > @hanzlamateen
> > 
> > 1. Install mkcert https://github.com/FiloSottile/mkcert
> > 2. Change .env.local CERT and KEY to `/certs/mkcert-cert.pem /certs/mkcert-key.pem`
> > 3. npm run dev-gencerts(Will use VITE_APP_HOST as the domain for the cert)
> > 4. Run npm run dev and enjoy a secure context
> 
> I tried following above steps and then tried opening engine in a new Edge browser window. It give me certificate error. Am I missing something?
> 
> image image

I think it's a possibility I am missing something. In this case I think I might need to conditionally pass the port if it is used :(

**Comment by DanielBelmes** [comment]
@hanzlamateen

Oh sorry I forgot to inform you that the .env has syntax errors that need to be updated. 

APP_TITLE=Ethereal Engine
to
APP_TITLE="Ethereal Engine"

and
SMTP_FROM_EMAIL=<ENTER_SENDER_EMAIL>
to
SMTP_FROM_EMAIL=ENTER_SENDER_EMAIL

**Comment by hanzlamateen** [comment]
Getting this behavior in Edge and Chrome. Btw, I am using WSL2.

https://github.com/EtherealEngine/etherealengine/assets/10975502/63f7b642-4494-473f-a84a-58315b612f96

**Comment by DanielBelmes** [comment]
@hanzlamateen ill look into a method to make it work for wsl. Since all the scripts are running in what I assume is your ubuntu vm. It won't update your windows local CA. I'm sure someone has tried this.

**Comment by aditya-mitra** [comment]
Hey Daniel,
Just asking, how did you solve the bug with local file server not shutting properly? I had also tried to kill the childProcess by id on main process exit but always got EADDRINUSE when the file server had to restart.

**Comment by DanielBelmes** [comment]
> Hey Daniel, Just asking, how did you solve the bug with local file server not shutting properly? I had also tried to kill the childProcess by id on main process exit but always got EADDRINUSE when the file server had to restart.

Just the work in packages/server-core/src/media/storageprovider/local.storage.ts

I'll make a seperate PR for it.

I found somewhere that someone ran the process detached and was able to sigint it. As much as i tried to kill on exit it never worked. But once I made it a detached process it finally worked. Must have something to do with node internals and how it allows the killing of a subprocess vs another process.

**Pull request closed without merging**